### DISCUSSION TITLE
Devices remain unavailable ater a network disconnect

### DISCUSSION THREAD

**Discussion by Brix1330** [make-all/tuya-local#1640] [Q&A]
So I was restarting my router, so wifi is temporarily off right... and wifi devices in HA obviously went unavailable. However after the wifi is up and running, Local Tuya devices remain unavailable, reloading them won't fix it. I had to restart HA in order to bring them back.

Has anyone experienced this? Any workarounds?

**Comment by make-all** [comment]
Did you set the IP address and protocol of your devices?

**Reply by Brix1330** [reply]
Yes, but the protocol, they're all auto.

**Reply by make-all** [reply]
Protocol auto means it has to cycle through all the protocols trying to find the right one.  Setting the protocol explicitly will help devices recover from connection problems.  If you need help figuring out which protocol your devices are using, download the diagnostics when the device is online and working, it will be mentioned in there.
